src/Build/Logging/BinaryLogger/BuildEventArgsReader.cs

```diff
@@ -258,9 +267,17 @@ private void ReadNameValueList()
                 $"NameValueList record number {recordNumber} is invalid: index {id} is not within {stringRecords.Count}.");
         }
 
+        private readonly StringReadEventArgs stringReadEventArgs = new StringReadEventArgs(string.Empty);
         private void ReadStringRecord()
         {
+            this.StringEncountered?.Invoke();
```

Comment:
Is this sufficient? Do you not also need it on ReadString and maybe GetStringFromRecord?

Reply:
Oh - good catch!
GetStringFromRecord is fine - as it only gets strings obtained via ReadStringRecord. But for v < 10 binlogs, the ReadString can be called in other places as well - so moved the hooks there